ProcessLib/ThermoMechanics/ThermoMechanicsFEM-impl.h

```diff
@@ -0,0 +1,278 @@
+void ThermoMechanicsLocalAssembler<ShapeFunction, IntegrationMethod,
+                                   DisplacementDim>::
+    assembleWithJacobianForHeatConductionEquations(
+        const double t, const std::vector<double>& /*local_xdot*/,
+        const double /*dxdot_dx*/, const double /*dx_dx*/,
+        std::vector<double>& /*local_M_data*/,
+        std::vector<double>& /*local_K_data*/,
+        std::vector<double>& local_b_data, std::vector<double>& local_Jac_data,
+        const LocalCoupledSolutions& local_coupled_solutions)
+{
+    auto const& local_T_vector =
+        local_coupled_solutions
+            .local_coupled_xs[_process_data.heat_conduction_process_id];
+    assert(local_T_vector.size() == temperature_size);
+    auto const local_T =
+        Eigen::Map<typename ShapeMatricesType::template VectorType<
+            temperature_size> const>(local_T_vector.data(), temperature_size);
+
+    auto const& local_T0_vector = local_coupled_solutions.local_coupled_xs0[0];
+    assert(local_T0_vector.size() == temperature_size);
+    auto const local_dT =
+        local_T -
+        Eigen::Map<typename ShapeMatricesType::template VectorType<
+            temperature_size> const>(local_T0_vector.data(), temperature_size);
+
+    auto local_Jac = MathLib::createZeroedMatrix<
+        typename ShapeMatricesType::template MatrixType<temperature_size,
+                                                        temperature_size>>(
+        local_Jac_data, temperature_size, temperature_size);
+
+    auto local_rhs = MathLib::createZeroedVector<
+        typename ShapeMatricesType::template VectorType<temperature_size>>(
+        local_b_data, temperature_size);
+
+    typename ShapeMatricesType::NodalMatrixType mass;
+    mass.setZero(temperature_size, temperature_size);
+
+    typename ShapeMatricesType::NodalMatrixType laplace;
+    laplace.setZero(temperature_size, temperature_size);
+
+    double const& dt = _process_data.dt;
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    ParameterLib::SpatialPosition x_position;
+    x_position.setElementID(_element.getID());
+
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        x_position.setIntegrationPoint(ip);
+        auto const& w = _ip_data[ip].integration_weight;
+        auto const& N = _ip_data[ip].N;
+        auto const& dNdx = _ip_data[ip].dNdx;
+
+        // calculate real density
+        // rho_s_{n+1} * (V_{n} + dV) = rho_s_n * V_n
+        // dV = 3 * alpha * dT * V_0
+        // rho_s_{n+1} = rho_s_n / (1 + 3 * alpha * dT )
+        // see reference solid density description for details.
+        auto& rho_s = _ip_data[ip].solid_density;
+        // calculate thermally induced strain
+        // assume isotropic thermal expansion
+        auto const alpha = _process_data.linear_thermal_expansion_coefficient(
+            t, x_position)[0];
+
+        double const dT = N.dot(local_dT);
```

Comment:
I'd think in other processes the standard notation was `dT_ip` for integration point value, and `dT` for part of `x_dot`. Same for the primary variables `T_ip` and `T`.

Reply:
Changed.